### ISSUE TITLE
Clearing execution time limit does not work

### ISSUE THREAD

**Issue by MetRonnie** [cylc/cylc-flow#5891]
### Description

Clearing the execution time limit either by editing `flow.cylc` and reloading, or by broadcast/edit-runtime, does not work. The original execution time limit is maintained.

### Reproducible Example

```ini
# flow.cylc
[scheduling]
    cycling mode = integer
    [[graph]]
        R1 = a
[runtime]
    [[a]]
        script = sleep 10
        execution time limit = PT1S
```

```console
$ cylc play workflow
```

Wait for task to fail.

```diff
     [[a]]
         script = sleep 10
-        execution time limit = PT1S
```

```console
$ cylc reload workflow
$ cylc trigger workflow//1/a
```

### Expected Behaviour

The task should succeed now after 10s, but it still fails after 1s

**Comment by wxtim** [comment]
I think I have it - `execution time limit = None` is entirely valid but [this code](https://github.com/cylc/cylc-flow/blob/b8a55d8f6ccfc57889f5e7b80e3baee5b8b613a4/cylc/flow/task_job_mgr.py#L1265-L1268) assumes that it isn't:

```
        with suppress(TypeError):
            itask.summary[self.KEY_EXECUTE_TIME_LIMIT] = float(
                rtconfig['execution time limit']
            )
```

**Comment by oliver-sanders** [comment]
Closed by #5902
